node/mock/peerProcessorMock.go

```diff
@@ -0,0 +1,75 @@
+package mock
+
+import (
+	"github.com/ElrondNetwork/elrond-go/data"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+)
+
+// ValidatorStatisticsProcessorMock -
+type ValidatorStatisticsProcessorMock struct {
+	UpdatePeerStateCalled func(header data.HeaderHandler) ([]byte, error)
+	RevertPeerStateCalled func(header data.HeaderHandler) error
+	IsInterfaceNilCalled  func() bool
+
+	GetPeerAccountCalled                     func(address []byte) (state.PeerAccountHandler, error)
+	RootHashCalled                           func() ([]byte, error)
+	ResetValidatorStatisticsAtNewEpochCalled func(vInfos map[uint32][]*state.ValidatorInfo) error
+	GetValidatorInfoForRootHashCalled        func(rootHash []byte) (map[uint32][]*state.ValidatorInfo, error)
+}
+
+// ResetValidatorStatisticsAtNewEpoch -
+func (vsp *ValidatorStatisticsProcessorMock) ResetValidatorStatisticsAtNewEpoch(vInfos map[uint32][]*state.ValidatorInfo) error {
+	if vsp.ResetValidatorStatisticsAtNewEpochCalled != nil {
+		return vsp.ResetValidatorStatisticsAtNewEpochCalled(vInfos)
+	}
+	return nil
+}
+
+// GetValidatorInfoForRootHash -
+func (vsp *ValidatorStatisticsProcessorMock) GetValidatorInfoForRootHash(rootHash []byte) (map[uint32][]*state.ValidatorInfo, error) {
+	if vsp.GetValidatorInfoForRootHashCalled != nil {
+		return vsp.GetValidatorInfoForRootHashCalled(rootHash)
+	}
+	return nil, nil
+}
+
+// UpdatePeerState -
+func (vsp *ValidatorStatisticsProcessorMock) UpdatePeerState(header data.HeaderHandler) ([]byte, error) {
+	if vsp.UpdatePeerStateCalled != nil {
+		return vsp.UpdatePeerStateCalled(header)
+	}
+	return nil, nil
+}
+
+// RevertPeerState -
+func (vsp *ValidatorStatisticsProcessorMock) RevertPeerState(header data.HeaderHandler) error {
+	if vsp.RevertPeerStateCalled != nil {
+		return vsp.RevertPeerStateCalled(header)
+	}
+	return nil
+}
+
+// RootHash -
+func (vsp *ValidatorStatisticsProcessorMock) RootHash() ([]byte, error) {
+	if vsp.RootHashCalled != nil {
+		return vsp.RootHashCalled()
+	}
+	return nil, nil
+}
+
+// GetPeerAccount -
+func (vsp *ValidatorStatisticsProcessorMock) GetPeerAccount(address []byte) (state.PeerAccountHandler, error) {
+	if vsp.GetPeerAccountCalled != nil {
+		return vsp.GetPeerAccountCalled(address)
+	}
+
+	return nil, nil
+}
+
+// IsInterfaceNil -
+func (vsp *ValidatorStatisticsProcessorMock) IsInterfaceNil() bool {
+	if vsp.IsInterfaceNilCalled != nil {
```

Comment:
return vsp==nil ? remove IsInterfaceNilCalled ?

Reply:
removed in multiple places